admin/pre-reg/database.py

```diff
@@ -0,0 +1,107 @@
+import httplib as http
```

Comment:
This file should be in its own PR.

A bit more to follow on the structure as I outline it, but I think this is either a "view" or a "model" but I think it is a view.